Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs

```diff
@@ -0,0 +1,355 @@
+            "FoodOatmeal",
+            "FoodBreadJellySlice",
+            "FoodMothCottonSalad",
+            "FoodBreadMoldy",
+            "FoodDonkpocketSpicyWarm",
+            "FoodCannabisButter",
+            "FoodNoodles",
+            "FoodBreadMeat",
+            "LeavesCannabisDried",
+            "FoodBurgerCheese",
+            "FoodDonkpocketDankWarm",
+            "FoodSpaceshroomCooked",
+            "FoodMealFries",
+            "MedicatedSuture",
+            "FoodDonkpocketWarm",
+            "FoodCakePlain",
+            "DisgustingSweptSoup",
+            "FoodBurgerPlain",
+            "FoodBreadGarlicSlice",
+            "FoodSoupMushroom",
+            "FoodSoupWingFangChu",
+            "FoodBreadMeatXeno",
+            "FoodCakeBrain",
+            "FoodBurgerBrain",
+            "FoodSaladCaesar"
+        };
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<RoboisseurComponent, ComponentInit>(OnInit);
+            SubscribeLocalEvent<RoboisseurComponent, InteractHandEvent>(OnInteractHand);
+            SubscribeLocalEvent<RoboisseurComponent, InteractUsingEvent>(OnInteractUsing);
+        }
+        private void OnInit(EntityUid uid, RoboisseurComponent component, ComponentInit args)
+        {
+            NextItem(component);
+        }
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+            foreach (var roboisseur in EntityQuery<RoboisseurComponent>())
+            {
+                roboisseur.Accumulator += frameTime;
+                roboisseur.BarkAccumulator += frameTime;
+                if (roboisseur.BarkAccumulator >= roboisseur.BarkTime.TotalSeconds)
+                {
+                    roboisseur.BarkAccumulator = 0;
+                    string message = Loc.GetString(_random.Pick(DemandMessages), ("item", roboisseur.DesiredPrototype.Name));
+                    if (roboisseur.ResetTime.TotalSeconds - roboisseur.Accumulator < 120)
+                    {
+                        roboisseur.Impatient = true;
+                        message = Loc.GetString(_random.Pick(ImpatientMessages), ("item", roboisseur.DesiredPrototype.Name));
+                    }
+                    else if (CheckTier(roboisseur.DesiredPrototype.ID) > 2)
+                        message = Loc.GetString(_random.Pick(DemandMessagesTier2), ("item", roboisseur.DesiredPrototype.Name));
+                    _chat.TrySendInGameICMessage(roboisseur.Owner, message, InGameICChatType.Speak, false);
+                }
+
+                if (roboisseur.Accumulator >= roboisseur.ResetTime.TotalSeconds)
+                {
+                    roboisseur.Impatient = false;
+                    NextItem(roboisseur);
+                }
+            }
+        }
+
+        private void RewardServicer(EntityUid uid, RoboisseurComponent component, int tier)
+        {
+            Random r = new Random();
```

Comment:
Haven't seen this Type before, I normally use the robust random system dependency.
```suggestion
            var r = new Random();
```